docs/source/nlp/text_normalization/wfst/wfst_customization.rst

```diff
@@ -0,0 +1,45 @@
+.. _wfst_customization:
+
+Grammar customization
+=====================
+
+.. note::
+
+    *TN/ITN is transitioning from [NVIDIA/NeMo](https://github.com/NVIDIA/NeMo) repository to a standalone [NVIDIA/NeMo-text-processing](https://github.com/NVIDIA/NeMo-text-processing) repository. All updates and discussions/issues should go to the new repository.*
+
+
+All grammar development is done with [Pynini library](https://www.opengrm.org/twiki/bin/view/GRM/Pynini).
+These grammars can be exported to .far files and used with Riva/Sparrowhawk, see :doc:`Text Processing Deployment <wfst_text_processing_deployment>` for details.
+
+Steps to customize grammars
+---------------------------
+
+1. Install [NeMo-TN from source](https://github.com/NVIDIA/NeMo-text-processing#from-source)
+2. Run [nemo_text_processing/text_normalization/normalize.py](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/normalize.py) or [nemo_text_processing/inverse_text_normalization/inverse_normalize.py](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/inverse_text_normalization/inverse_normalize.py) with `--verbose` flag to evaluate current behavior on the target case, see argument details in the scripts and [this tutorial](https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/text_processing/Text_(Inverse)_Normalization.ipynb)
+3. [This function](https://github.com/kylebgorman/pynini/blob/master/pynini/lib/rewrite.py#L292) might be helpful for grammar development/debugging
+
+.. code-block:: python
+
+    from pynini.lib.rewrite import top_rewrite
+    print(top_rewrite(input_string, graph))
```

Comment:
What does this print?

Reply:
Moreover point 3 doesn't make much sense to me at the moment

Reply:
this is meant for debugging purposes, just one of the ways to evaluate if the graph is working for the input_string. I combined 3 and 4 into a single step, more details in the linked tutorial